library/alloc/src/string.rs

```diff
@@ -1227,17 +1227,11 @@ impl String {
     where
         P: for<'x> Pattern<'x>,
     {
-        use core::str::pattern::Searcher;
+        use core::str::pattern::Searcher as _;
```

Comment:
Is this necessary?

Reply:
No. Style choice - brings in the trait methods without bringing in the `Searcher` item.

Reply:
Reverted.